Square_Root_Decomposition/Square_Root_Decomposition.cpp

```diff
@@ -0,0 +1,61 @@
+#include<bits/stdc++.h>
+using namespace std;
+
+void update(int index,int value,int *blocks,int a[],int n){
+	blocks[(int)(index/sqrt(n))] += value-a[index];
+	a[index]=value;
+}
+//assuming 0 based indexing
+int query(int low,int high,int *blocks,int a[],int n){
+	int block_size=sqrt(n);
+	int sum=0;
+
+	while(low % block_size!=0 && low<high){
+		sum+=a[low];
+		low++;
+	}
+	while(low+block_size<=high){
+		sum+=blocks[low/block_size];
+		low+=block_size;
+	}
+	while(low<=high){
+		sum+=a[low];
+		low++;
+	}
+
+	return sum;
+
+}
+//divide the array in blocks of size sqrt(n), sum of elements in each block is stored in blocks
+int *formBlocks(int a[],int n){
+	int block_size=sqrt(n);
+	int index=-1;
+	int *blocks = new int[(int)(n/block_size)+1];
+
+	for(int i=0;i<n;i++){
+		if(i%block_size==0){
+			index++;
+		}
+		blocks[index]+=a[i];
+	}
+	return blocks;
+}
+int main(){
+	/* Applying square root decomposition to perform range queries
+	for finding the sum of numbers in a given range 
+	Indexing : 0 based in array
+	*/
+	int n = 10; //number of elements 
```

Comment:
always better to calculate the size of the array by using sizeof(arr)/sizeof(arr[0])

Reply:
I have updated the code